lib/Mojo/Util.pm

```diff
@@ -88,6 +88,33 @@ monkey_patch(__PACKAGE__, 'sha1_sum',      \&sha1_hex);
 monkey_patch(__PACKAGE__, 'steady_time',
   MONOTONIC ? sub () { Time::HiRes::clock_gettime(Time::HiRes::CLOCK_MONOTONIC()) } : \&Time::HiRes::time);
 
+# Building random bytes function
+monkey_patch(__PACKAGE__, 'urandom_bytes', $^O eq 'MSWin32' ? _build_urandom_bytes_win32() : _build_urandom_bytes());
+
+sub _build_urandom_bytes_win32 {
+  require Win32::API;
+
+  my $RtlGenRandom = Win32::API->new('advapi32', 'INT SystemFunction036 (PVOID RandomBuffer, ULONG RandomBufferLength)')
+    or croak "SystemFunction036 import failed: $!";
+
+  return sub {
+    my $num   = shift || 64;
+    my $bytes = chr(0) x $num;
+    $RtlGenRandom->Call($bytes, $num) or croak "SystemFunction036 call failed: $!";
+    return $bytes;
+  }
+}
+
+sub _build_urandom_bytes {
+  open(my $urandom, '<', '/dev/urandom') or croak "Cannot open /dev/urandom: $!";
+
+  return sub {
+    my $num = shift || 64;
+    sysread($urandom, my $bytes, $num);
```

Comment:
Is it worth checking the return to find out of we got all the bytes we asked for?

Reply:
This is covered by tests, I'm not sure how necessary it is to assert at runtime but it would be a lightweight check.

Reply:
A simple check for this has been added:

```perl
    sysread($urandom, $bytes, $num) == $num or die "sysread() from /dev/urandom didn't return $num bytes";

```